source-oracle-batch/main_test.go

```diff
@@ -73,13 +73,81 @@ func TestBasicCapture(t *testing.T) {
 		// Run the capture over and over for 5 seconds each time until all inserts have finished, then verify results.
 		for !insertsDone.Load() {
 			var captureCtx, cancelCapture = context.WithCancel(ctx)
-			time.AfterFunc(6*time.Second, cancelCapture)
+			time.AfterFunc(5*time.Second, cancelCapture)
 			cs.Capture(captureCtx, t, nil)
 		}
 		cupaloy.SnapshotT(t, cs.Summary())
 	})
 }
 
+type doc struct {
+	Id string `json:"ID"`
+}
+
+func TestCaptureCheckpointSCN(t *testing.T) {
```

Comment:
I tested this behavior by setting `scnCursor=false` and running the test and seeing it fail, while with `scnCursor=true` it works 👍🏽